eos/scattering/observables.cc

```diff
@@ -0,0 +1,224 @@
+                        Unit::GeV(),
+                        &EEToCCBar::Jpsi_total_width),
+
+                make_observable("psi(2S)::total_width", R"($\Gamma_{\psi(2S)}$)",
+                        Unit::GeV(),
+                        &EEToCCBar::psi2S_total_width),
+
+                make_observable("psi(3770)->D^0Dbar^0::decay_width", R"($\Gamma(\psi(3770) \to D^0\bar{D}^0)$)",
+                        Unit::GeV(),
+                        &EEToCCBar::psi3770_D0Dbar0_width),
+
+                make_observable("psi(3770)->D^+D^-::decay_width", R"($\Gamma(\psi(3770) \to D^+\D^-)$)",
+                        Unit::GeV(),
+                        &EEToCCBar::psi3770_DpDm_width),
+
+                make_observable("psi(3770)->eff::decay_width", R"($\Gamma(\psi(3770) \to \textrm{eff})$)",
+                        Unit::GeV(),
+                        &EEToCCBar::psi3770_eff_width),
+
+                make_observable("psi(3770)::total_width", R"($\Gamma_{\psi(3770)}$)",
+                        Unit::GeV(),
+                        &EEToCCBar::psi3770_total_width),
+
+                make_observable("psi(3770)::spectral_function(E)", R"($\textrm{spect}_{\psi(3770)}(E)$)",
+                        Unit::InverseGeV2(),
+                        &EEToCCBar::psi3770_spectral_function,
+                        std::make_tuple("E")),
+
+                make_cacheable_observable("e^+e^-->e^+e^-::sigma(E)", R"($\sigma(e^+e^- \to e^+e^-)|_{s\textrm{-channel}}$)",
+                        Unit::InverseGeV2(),
+                        &EEToCCBar::prepare,
+                        &EEToCCBar::sigma_eetoee,
+                        std::make_tuple("E")),
+
+                make_cacheable_observable("e^+e^-->eff::sigma(E)", R"($\sigma(e^+e^- \to \textrm{eff})$)",
+                        Unit::InverseGeV2(),
+                        &EEToCCBar::prepare,
+                        &EEToCCBar::sigma_eetoeff,
+                        std::make_tuple("E")),
+
+                make_cacheable_observable("e^+e^-->D^0Dbar^0::sigma(E)", R"($\sigma(e^+e^- \to D^0 \bar{D}^0)$)",
+                        Unit::InverseGeV2(),
+                        &EEToCCBar::prepare,
+                        &EEToCCBar::sigma_eetoD0Dbar0,
+                        std::make_tuple("E")),
+
+                make_cacheable_observable("e^+e^-->D^+D^-::sigma(E)", R"($\sigma(e^+e^- \to D^+ D^-)$)",
+                        Unit::InverseGeV2(),
+                        &EEToCCBar::prepare,
+                        &EEToCCBar::sigma_eetoDpDm,
+                        std::make_tuple("E")),
+
+                make_cacheable_observable("e^+e^-->ccbar::R(E)", R"($R$)",
+                        Unit::None(),
+                        &EEToCCBar::prepare,
+                        &EEToCCBar::R,
+                        std::make_tuple("E")),
+
+                // Phase space factor
+                make_cacheable_observable("e^+e^-::re_rho(re(E),im(E))", R"($\mathrm{Re}\rho_{e^+e^-}(E))$)",
```

Comment:
I think typically we write ``Re{..}`` and ``Im{..}``. So should this be ``Re{rho}(Re{E},Im{E})``?

Reply:
Also, I'm not too happy about the qualified name's prefix. This is all related to ``ee->ccbar``, right? So we should probably keep everything within that namespace. What happens if we add the scattering of e^+e^-->pipi in the future?

Reply:
No, this is a property of the channel, it is unrelated to the process

Reply:
Fair enough.